### PR TITLE
Manual backport of #11250 (Ensure that the CDS unit parser can deal with '[-]')

### PR DESCRIPTION
**Pull request by pllim**
Ensure that the CDS unit parser can deal with '[-]'

### Description

This pull request is to manually backport #11250

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -1152,6 +1152,9 @@ astropy.units
 - Ensure ``keepdims`` works for taking ``mean``, ``std``, and ``var`` of
   ``Quantity``. [#11198]
 
+- For CDS units and tables, recognize ``-`` as indicating dimensionless and
```
Comment (pllim) [inline CHANGES.rst:1155]: Do I have to move the change log?

**Comment by mhvk** [comment]
The backport looks good (except this time I think the changelog should be moved as it really won't be in 4.0 any more - in fact, shouldn't it be moved in master if we do not end up backporting to 4.0? let me comment there too)

I don't understand the failures, though... They seem totally unrelated.

**Comment by pllim** [comment]
The failures are from #11305 , I think?

**Comment by mhvk** [comment]
Darn, reminds me that we need a pyerfa 1.7.3...

**Comment by eteq** [comment]
Yeah, I agree the failures are unrelated.  To finish backporting I'm going to merge this, but I think we need to either xfail or make sure 1.7.3 is out first (and exclude 1.7.2)

**Pull request merged**